src/app/mailviewer/rmm7messageactions.ts

```diff
@@ -17,43 +17,76 @@
 // along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
 // ---------- END RUNBOX LICENSE ----------
 
+import { Observable, of } from 'rxjs';
 import { DraftDeskService, DraftFormModel } from '../compose/draftdesk.service';
 import { SingleMailViewerComponent } from './singlemailviewer.component';
 import { MoveMessageDialogComponent } from '../actions/movemessage.action';
 import { SearchService } from '../xapian/searchservice';
+import { MessageListService } from '../rmmapi/messagelist.service';
 import { MatDialog } from '@angular/material/dialog';
 import { MatSnackBar } from '@angular/material/snack-bar';
 import { MessageActions } from './messageactions';
-import { RunboxWebmailAPI } from '../rmmapi/rbwebmail';
+import { RunboxWebmailAPI, MessageFlagChange } from '../rmmapi/rbwebmail';
 import { ProgressDialog } from '../dialog/dialog.module';
 
 export class RMM7MessageActions implements MessageActions {
     mailViewerComponent: SingleMailViewerComponent;
     dialog: MatDialog;
     snackBar: MatSnackBar;
     searchService: SearchService;
+    messageListService: MessageListService;
     draftDeskService: DraftDeskService;
     rmmapi: RunboxWebmailAPI;
 
+    public updateMessages(args: { messageIds: number[],
```

Comment:
This type definition has become a bit bulky, maybe we can just name a type after it? :)